src/Gauge/NavigatorWidget.cpp

```diff
@@ -0,0 +1,234 @@
+/*
+  Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "NavigatorWidget.hpp"
+#include "UIGlobals.hpp"
+#include "Look/Look.hpp"
+#include "Interface.hpp"
+#include "ui/window/AntiFlickerWindow.hpp"
+#include "ui/canvas/Canvas.hpp"
+#include "Renderer/NavigatorRenderer.hpp"
+#include "Screen/Layout.hpp"
+
+#include <iostream>
+
+/**
+ * A Window which renders a Navigator
+ */
+class NavigatorWindow : public AntiFlickerWindow {
+  const NavigatorLook &look;
+  const TaskLook &look_task;
+  const bool& inverse;
+
+  AttitudeState attitude;
+  
+public:
+  /**
+   * Constructor. Initializes most class members.
+   */
+  NavigatorWindow(const NavigatorLook &_look, const TaskLook &_look_task, const bool &_inverse) noexcept
+    :look(_look), look_task(_look_task), inverse(_inverse) {
+    attitude.Reset();
+  }
+
+  void ReadBlackboard(const AttitudeState _attitude) noexcept {
+    attitude = _attitude;
+    Invalidate();
+  }
+
+protected:
+  /* virtual methods from AntiFlickerWindow */
+  void OnPaintBuffer(Canvas &canvas) noexcept override {
+    if (inverse)
+      canvas.Clear(COLOR_BLACK);
+    else
+      canvas.ClearWhite();
+
+    // if (!attitude.bank_angle_available && !attitude.pitch_angle_available) {
+    //   NavigatorRenderer::DrawText(canvas, canvas.GetRect(), look);
+    //   return;
+    // }
+
+    const auto &basic = CommonInterface::Basic();
+    TimeStamp current_time{};
```

Comment:
What's the point of this initaliser?

Reply:
No more used; It was used during 'debugging/learning time'. current_time variable and followings deleted.